### ISSUE TITLE
Regression when use  containers with multiple labels

### ISSUE THREAD

**Issue by DirkTheDaring** [containous/traefik#2630]
## What did you expect to see?
My docker apps running in traefik with several containers using multiple routing like the example below.
This works with my old 1.3.0-rc3  image , tried today to upgrade to latest, then it broke. When i reverted back to the old image, all good again.

This is part of a service running in docker compose.

labels:
  - traefik.servicename1.frontend.rule=Host:example1.com
  - traefik.servicename2.frontend.rule=Host:example2.com

### What did you see instead?
When starting up, for the containers with multiple labels as up there errors started to appear.
level=error msg="Undefined backend .... '   and the traefik routing not working.
The whole string looks very weird then as it refers to an internal name in docker.

### Output of `traefik version`: (_What version of Traefik are you using?_)

 v1.5.0-rc3 - pulled traefik: v1.5.0-rc3  container

Release v1.5.0-rc3 + 1.4.4 + 1.4.5

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
traefik runs itself runs in the docker container.

```toml
debug=true
logLevel="ERROR"
defaultEntryPoints = ["http", "https"]


[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
   [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "/etc/pki/tls/certs/example1.com.crt"
      keyFile = "/etc/pki/tls/private/example1.com.key"
      [[entryPoints.https.tls.certificates]]
      certFile = "/etc/letsencrypt/live/example2.com.pem"
      keyFile =  "/etc/letsencrypt/live/example2.com/privkey.pem"


[web]
address = ":8080"
CertFile = "/etc/pki/tls/certs/example1.com.crt"
KeyFile =  "/etc/pki/tls/private/example1.com.key"
[docker]
endpoint = "unix:///var/run/docker.sock"
```

**Comment by mmatur** [comment]
@DirkTheDaring Could you send us more informations about your compose file please

**Comment by mmatur** [comment]
@DirkTheDaring Could you send us your full Træfik log please ?

**Comment by DirkTheDaring** [comment]
Sure here is a sample which reproduces the bug in my enviroment.
Only hostnames are exchanged due to privacy reasons.
---
version: '2'

services:
  traefik:
          #image: traefik:1.3.0-rc3
          #image: traefik:1.4.0
    image: traefik:v1.5.0-rc3
    restart: unless-stopped
    container_name: traefik.prod.cloud
    hostname: traefik

    volumes:
    - /var/run/docker.sock:/var/run/docker.sock
    - /etc/traefik:/etc/traefik

    ports:
    - 80:80

  nginx:
    image: nginx
    restart: unless-stopped
    container_name: nginx.prod.cloud
    hostname: nginx

    ports:
    - 80:80

    labels:
    - traefik.servicename1.frontend.priority=10
    - traefik.servicename1.frontend.rule=Host:example1.com
    - traefik.servicename1.port=80
    - traefik.servicename2.frontend.priority=10
    - traefik.servicename2.frontend.rule=Host:example2.com
    - traefik.servicename2.port=80

**Comment by DirkTheDaring** [comment]
also 1.4.0 works for me.

**Comment by DirkTheDaring** [comment]
triaged the introduction of the bug to  1.4.4  .   It works until traefik:1.4.3  container. Then breaks on 1.4.4 and stays broken (1.4.5) until then.

**Comment by mmatur** [comment]
@DirkTheDaring Thanks for your investigation. 

This regression is due to this PR #2434 

We are working on a fix

**Comment by traefiker** [comment]
Closed by #2631.
